tests/io/delta_lake/test_table_write.py

```diff
@@ -180,3 +182,61 @@ def test_deltalake_write_ignore(tmp_path):
     expected_schema = Schema.from_pyarrow_schema(read_delta.schema().to_pyarrow())
     assert df1.schema() == expected_schema
     assert read_delta.to_pyarrow_table() == df1.to_arrow()
+
+
+@pytest.mark.parametrize(
+    "partition_cols,num_partitions",
+    [
+        (["int"], 2),
+        (["float"], 2),
+        (["str"], 2),
+        pytest.param(["bin"], 2, marks=pytest.mark.xfail(reason="Binary partitioning is not yet supported")),
+        (["bool"], 2),
+        (["datetime"], 2),
+        (["date"], 2),
+        (["decimal"], 2),
+        (["int", "float"], 3),
+    ],
+)
+def test_deltalake_write_partitioned(tmp_path, partition_cols, num_partitions):
```

Comment:
Thoughts on other potential corner-cases? Things like empty tables, `None` values for the weird `__HIVE_DEFAULT_PARTITION__` fallback etc.

Reply:
Will add more tests 👍